clang/lib/Sema/SemaExpr.cpp

```diff
@@ -12881,6 +12886,12 @@ QualType Sema::CheckVectorLogicalOperands(ExprResult &LHS, ExprResult &RHS,
   if (!getLangOpts().CPlusPlus &&
       !(isa<ExtVectorType>(vType->getAs<VectorType>())))
     return InvalidLogicalVectorOperands(Loc, LHS, RHS);
+  if (getLangOpts().HLSL &&
+      getLangOpts().getHLSLVersion() >= LangOptionsBase::HLSL_2021) {
+    (void)InvalidOperands(Loc, LHS, RHS);
```

Comment:
Throwing away the return value here makes me uneasy, but as long as you know that `InvalidOperands` will _always_ return `QualType()` it doesn't matter so much.  I assume people working on this code have that pretty internalized as an idomatic convention.